sensor/kubernetes/listener/resources/deployments.go

```diff
@@ -153,18 +159,28 @@ func (d *deploymentHandler) processWithType(obj, oldObj interface{}, action cent
 					owningDeploymentIDs.AsSlice(), uid, objAsPod.Namespace, objAsPod.Name)
 			}
 		}
+
+		var removeEvents *central.SensorEvent
 		// On removes, we may not get the owning deployment ID if the deployment was deleted before the pod.
 		// This is okay. We still want to send the remove event anyway.
 		if action == central.ResourceAction_REMOVE_RESOURCE || owningDeploymentID != "" {
-			events = component.MergeResourceEvents(events, d.processPodEvent(owningDeploymentID, objAsPod, action))
+			removeEvents = d.processPodEvent(owningDeploymentID, objAsPod, action)
 		}
-	}
 
-	if deploymentWrap == nil {
-		if objAsPod != nil {
+		if deploymentWrap == nil {
 			// It's only a pod event and the pod belongs to another resource (e.g. deployment)
-			events = component.MergeResourceEvents(events, d.maybeUpdateParentsOfPod(objAsPod, oldObj, action))
+			d.maybeUpdateParentsOfPod(objAsPod, oldObj, action, events)
+			if removeEvents != nil {
+				events.AppendMessage(removeEvents)
```

Comment:
[nit] Would move this to line 167 (right after calling `processPodEvent`) make sense? It looks like if `removeEvents` is not `nil` we are appending the events always anyway. Unless `maybeUpdateParentsOfPod` should be called before appending, but I don't think the order matters here 🤔